hpack/hpack.py

```diff
@@ -231,19 +234,17 @@ def encode(self, headers, huffman=True):
         if isinstance(headers, dict):
             headers = headers.items()
 
-        # Next, walk across the headers and turn them all into bytestrings.
-        headers = [(_to_bytes(n), _to_bytes(v)) for n, v in headers]
-
         # Before we begin, if the header table size has been changed we need
         # to signal that appropriately.
         if self._table_size_changed:
             header_block.append(self._encode_table_size_change())
             self._table_size_changed = False
 
-        # We can now encode each header in the block.
-        header_block.extend(
-            (self.add(header, huffman) for header in headers)
-        )
+        # Add each header to the header block
+        for header in headers:
+            sensitive = header[2] if len(header) > 2 else False
+            header = (_to_bytes(header[0]), _to_bytes(header[1]))
+            header_block.extend([self.add(header, sensitive, huffman)])
```

Comment:
Ok, this creates an unfortunate temporary list. Rather than do `header_block.extend([self.add(header, sensitive, huffman)])`, let's do `header_block.append(self.add(header, sensitive, huffman))`.